crates/volta-core/src/platform/sourced.rs

```diff
@@ -8,8 +8,14 @@ use semver::Version;
 use volta_fail::Fallible;
 
 pub enum Source {
+    /// Represents a Platform that came from the user default
     User,
+
+    /// Represents a Platform that came from a project manifest
     Project,
+
+    /// Represents a Platform that is the result of merging the User and Project platforms
+    ProjectNodeUserYarn,
```

Comment:
Do we need to support the *inverse* of this? I.e. `ProjectYarnUserNode` seems like a thing that might exist, even if we wouldn't recommend it. In that case, it seems like we might want to restructure this slightly (perhaps in a follow-on PR; I can open an issue) to differentiate the source of *each tool* rather than the source of the whole platform. As it turns out, that's also [the tack I've ended up needing to take](https://github.com/volta-cli/volta/blob/c56fcab253cece17772a62a955f9a1a68cd1ae27/src/command/list/mod.rs#L54:L77) simply to be able to accurately *describe* the user's toolchain in `volta list`.

Reply:
We don't (currently) need to support`ProjectYarnUserNode`, as we require the `node` setting to be the start of any platform. So if a project has `yarn`, then it must necessarily have `node` set as well (or we would have returned an error on the parse).

That said, I agree that having the source of each tool separately makes sense, and I actually started with the code organized that way. This PR, along with #466, were the result of thinking about the different issues (this one, the verbose logging for tools, and the work you're doing on `volta list`) and realizing that there was a need for some sort of "Source" attached to our `PlatformSpec` object. I spent an inordinate amount of time trying to work through a bunch of conceptual and technical hurdles, and while I'm definitely not thrilled with the final result, it was the least bad that I could come up with.

That said, I'm open to doing some more brainstorming on how to track / propagate the source through the data structures in a cleaner way, perhaps we can touch on that in the open call today, or chat on Discord.

Reply:
Least bad is something with which I sympathize deeply and it's much how I feel about this space as well. Do let's chat; but let's go ahead and land this as is!